### PR TITLE
Add accessors to model and unit tests, Javadoc

### PR DESCRIPTION
**Pull request by nbauma109**
- Added accessors to the module classes. Until now, the fields could only be accessed by reflection and by calling setAccessible(true). The getters return the values from the constant pool as opposed to indexes for more convenient usage.
- Added getter for the local variable type table, similarly to local variable table. They go together, it makes sense to have both instead of one of them.
- Added getAttribute to search attribute by tag. It will avoid iterating through attributes find an attribute or streaming the attributes like this Stream.of(getAttributes()).filter(MyAttribute.class::isInstance).findFirst()

### CODE REVIEW

**Comment by nbauma109** [comment]
Unable to resolve action `actions/setup-java@v3.7.0`, unable to find version `v3.7.0`

**Comment by nbauma109** [comment]
Hi @garydgregory 
Please help moving forward with this PR
Thanks !

**Comment by garydgregory** [comment]
This PR has no description, why is it needed?

File: src/main/java/org/apache/bcel/classfile/Code.java [hunk 1]
```diff
@@ -259,6 +259,20 @@ public LocalVariableTable getLocalVariableTable() {
         return null;
     }
 
+    /**
+     * Gets the local variable type table attribute {@link LocalVariableTypeTable}.
+     * @return LocalVariableTypeTable of Code, if it has one, null otherwise.
+     * @since 6.7.1
```
Comment (garydgregory) [inline src/main/java/org/apache/bcel/classfile/Code.java:265]: New Javadoc since tags should be for 6.8.0.

**Review by garydgregory — CHANGES_REQUESTED** [review]
That's a lot of new APIs for one PR, are ALL of these new APIs actually used? If not, let's split this up since a PR where APIs are actually used from main, and another for the other APIs, otherwise this feels like we would be adding APIs for the sake of adding APIs. TY!

**Comment by nbauma109** [comment]
> That's a lot of new APIs for one PR, are ALL of these new APIs actually used? If not, let's split this up since a PR where APIs are actually used from main, and another for the other APIs, otherwise this feels like we would be adding APIs for the sake of adding APIs. TY!

I want to rewrite a cleaner code and avoid reflection hacks (i.e., setAccessible(true)), example here:
https://github.com/nbauma109/jd-core/blob/0bcbf35f700f0cc48e1ef79456c37f9168f527ff/src/main/java/org/jd/core/v1/service/converter/classfiletojavasyntax/processor/ConvertClassFileProcessor.java#L375

I want to avoid recycling this boilerplate pattern for every single attribute:

```java
    MyAttribute attr = (MyAttribute) Stream.of(fieldOrMethod.getAttributes()).filter(MyAttribute.class::isInstance).findAny().orElse(null);
```

I have lots of them in that same class ConvertClassFileProcessor and in LocalVariableMaker.

Other example for local variable type table:
https://github.com/nbauma109/jd-core/blob/0bcbf35f700f0cc48e1ef79456c37f9168f527ff/src/main/java/org/jd/core/v1/service/converter/classfiletojavasyntax/util/ByteCodeWriter.java#L525

Also not sure what you mean by "are these new APIs actually used?", as far as I understand, they can't be used until they're published. I happen to create SNAPSHOTs of my forked BCEL for myself to check everything works as I want but that's it.

File: src/main/java/org/apache/bcel/classfile/Code.java (hunk 1 above)
Comment (nbauma109) [reply src/main/java/org/apache/bcel/classfile/Code.java]: done

**Comment by garydgregory** [comment]
@nbauma109 
Thank you for the reply and please accept my apologies for the delay, I'll come back around to this PR this week.

**Comment by nbauma109** [comment]
Hi, any chance to get this in ? What version should I use for the @since tags ?

**Comment by garydgregory** [comment]
Hi @nbauma109 

I apologize for the delay.

- The next version for new APIs should be `@since 6.10.0`
- Please rebase on git master
- Run `mvn` (by itself) and make sure the build passes. It may not if there are not enough tests to cover the new code. All new public and protected methods should have matching tests

**Comment by nbauma109** [comment]
Hi @garydgregory 

Can you enable the workflows ?

Thanks.

**Comment by garydgregory** [comment]
Hi @markro49 
Would you have any issues on your end with this PR? Looks good to me.

**Review by garydgregory — APPROVED** [review]
LGTM

**Review by markro49 — APPROVED** [review]

**Comment by garydgregory** [comment]
Merged, TY @markro49 !

**Pull request merged**